reference/artifacts/in-pipelines/index.md

```diff
@@ -0,0 +1,137 @@
+running.
+
+{% include figure
+   image_path="./expected-artifact-trigger.png"
+   caption="The UI provides short-hand for defining some types of
+            artifacts, in this case a docker image. This is optional, but helps
+            quickly define common types of artifacts."
+%}
+
+In this case, it's possible to define fallback behavior for when this artifact
+isn't bound at the start of the Pipeline. There are two options, evaluated in
+order:
+
+1. __Use Prior Execution__
+
+   When selected, whatever artifact was bound by this artifact in the
+   pipeline's prior execution is used. This is useful in cases where multiple
+   sources can trigger the pipeline, but each has incomplete information about
+   what artifacts to supply.
+
+2. __Use Default Artifact__
+
+   When selected, you can define by hand exactly which artifact to bind.
+
+Once you have declared which artifacts are expected by this pipeline, you can
+assign expected artifacts to individual triggers.
+
+{% include figure
+   image_path="./pubsub-trigger-artifact.png"
+   caption="Above is a pubsub subscription configured to listen to changes in
+            one GCR registry. Since this registry can contain many
+            repositories, we've assigned it an expected artifact to ensure only
+            changes in one repository can run this pipeline."
+%}
+
+When a trigger has one or more expected artifacts, it only runs when each
+expected artifact can bind to one of artifacts in the trigger's payload.
+
+### Artifacts in Trigger Payloads
+
+Artifacts are supplied by payload as a list of artifacts in a top-level
+`artifacts` key - the value is automatically injected into any triggered
+pipeline's execution context. However, it's possible that you're not the author
+of the incoming message, and are instead depending on a third-party system to
+provide an event to Spinnaker to trigger a pipeline. If this event contains
+information such as "these images were created", or "these files were
+modified", it's useful to be able to extract artifacts from that payload.
+
+For this reason, we allow you to supply [Jinja
+templates](http://jinja.pocoo.org/) that transform the incoming payload into a
+list of artifacts to be injected into your pipeline.
+
+Take for example the [notification format for
+GCR](https://cloud.google.com/container-registry/docs/configuring-notifications#notification_examples).
+While it captures the information we need, it's not in the format we want. So,
+we can register the following Jinja template for our GCR subscription:
+
+```json
+[
+  {
+    {% raw %}
+    {% set split = digest.split("@") %}
+    "reference": "{{ digest }}",
+    "name": "{{ split[0] }}",
+    "version": "{{ split[1] }}",
+    "type": "docker/image"
+    {% endraw %}
+  }
+]
+```
+
+## Find Artifact from Execution
+
+To allow you to promote artifacts between executions, you can make use of the
+"Find Artifact from Execution" stage. All that's required is the ID of the
```

Comment:
Is this the execution ID or the pipeline name?

Reply:
pipeline ID, clearing up the wording...